internal/api/documents_related_resources.go

```diff
@@ -0,0 +1,263 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp-forge/hermes/pkg/algolia"
+	hcd "github.com/hashicorp-forge/hermes/pkg/hashicorpdocs"
+	"github.com/hashicorp-forge/hermes/pkg/models"
+	"github.com/hashicorp/go-hclog"
+	"gorm.io/gorm"
+)
+
+type relatedResourcesPutRequest struct {
+	ExternalLinks   []externalLinkRelatedResourcePutRequest   `json:"externalLinks,omitempty"`
+	HermesDocuments []hermesDocumentRelatedResourcePutRequest `json:"hermesDocuments,omitempty"`
+}
+
+type externalLinkRelatedResourcePutRequest struct {
+	Name      string `json:"name"`
+	URL       string `json:"url"`
+	SortOrder int    `json:"sortOrder"`
+}
+
+type hermesDocumentRelatedResourcePutRequest struct {
+	GoogleFileID string `json:"googleFileID"`
+	SortOrder    int    `json:"sortOrder"`
+}
+
+type relatedResourcesGetResponse struct {
+	ExternalLinks   []externalLinkRelatedResourceGetResponse   `json:"externalLinks,omitempty"`
+	HermesDocuments []hermesDocumentRelatedResourceGetResponse `json:"hermesDocuments,omitempty"`
+}
+
+type externalLinkRelatedResourceGetResponse struct {
+	Name      string `json:"name"`
+	URL       string `json:"url"`
+	SortOrder int    `json:"sortOrder"`
+}
+
+type hermesDocumentRelatedResourceGetResponse struct {
+	GoogleFileID   string `json:"googleFileID"`
+	Title          string `json:"title"`
+	DocumentType   string `json:"documentType"`
+	DocumentNumber string `json:"documentNumber"`
+	SortOrder      int    `json:"sortOrder"`
+}
+
+func documentsResourceRelatedResourcesHandler(
+	w http.ResponseWriter,
+	r *http.Request,
+	docID string,
+	docObj hcd.Doc,
+	l hclog.Logger,
+	algoRead *algolia.Client,
+	db *gorm.DB,
+) {
+	switch r.Method {
+	case "GET":
+		d := models.Document{
+			GoogleFileID: docID,
+		}
+		if err := d.Get(db); err != nil {
+			l.Error("error getting document from database",
+				"error", err,
+				"path", r.URL.Path,
+				"method", r.Method,
+				"doc_id", docID,
+			)
+			http.Error(w, "Error accessing document",
```

Comment:
Maybe a more specific error would be warranted in this route as the user facing error?

```suggestion
			http.Error(w, "Error accessing document resources",
```

I don't have a strong opinion on this but wanted to point it out.